src/ng/browser.js

```diff
@@ -79,6 +83,19 @@ function Browser(window, document, $log, $sniffer) {
     }
   };
 
+  self.shutdown = function() {
+    active = false;
+    forEach(currentIntervalIds, function(ignore, intervalId) {
+      delete currentIntervalIds[intervalId];
+      clearInterval(+intervalId);
+    });
+    forEach(pendingDeferIds, function(ignore, timeoutId) {
+      delete pendingDeferIds[timeoutId];
+      clearTimeout(+timeoutId);
+    });
+    jqLite(window).off('hashchange popstate', cacheStateAndFireUrlChange);
```

Comment:
For that to work we first need to fix #13571 (see this comment in particular: https://github.com/angular/angular.js/issues/13571#issuecomment-165771088).